crates/cli/src/config.rs

```diff
@@ -511,15 +586,76 @@ impl Config {
         Self {
             home: RawConfig::new_with_localhost(),
             home_path,
+            doc: Default::default(),
         }
     }
 
+    /// Returns a preserving copy of [Config].
```

Comment:
Preserving what? Also, use backticks when linking to code names in doc comments:

```suggestion
    /// Returns a preserving copy of [`Config`].
```